src/cephadm/cephadm

```diff
@@ -627,26 +627,27 @@ def infer_fsid(func):
     @wraps(func)
     def _infer_fsid():
         if args.fsid:
-            logger.debug('Using specified fsid: %s' % args.fsid)
+            logger.info('Using specified fsid: %s' % args.fsid)
             return func()
 
-        fsid_list = []
-        if os.path.exists(args.data_dir):
-            for i in os.listdir(args.data_dir):
-                if is_fsid(i):
-                    fsid_list.append(i)
-        logger.debug('Found fsids %s' % str(fsid_list))
-
-        if not fsid_list:
-            # TODO: raise?
-            return func()
-
-        if len(fsid_list) > 1:
+        fsids = set()
+        daemon_list = list_daemons(detail=False)
+        for daemon in daemon_list:
+            if 'name' not in args or not args.name:
+                fsids.add(daemon['fsid'])
+            elif daemon['name'] == args.name:
+                fsids.add(daemon['fsid'])
+        fsids = list(fsids)
+        logger.info('Found fsids %s' % str(fsids))
+
+        if len(fsids) == 1:
+            logger.info('Inferring fsid %s' % fsids[0])
+            args.fsid = fsids[0]
+        elif len(fsids) > 1:
```

Comment:
Can we please also handle the case where `len(fsid) == 0`? Otherwise this looks like a bug to everyone.
```suggestion
        else:
```

Reply:
I added a noop branch with comment for the `len(fsid) == 0` case. Better?